tasks/sep/026-fp-holocene-upgrade-fix/VALIDATION.md

```diff
@@ -0,0 +1,49 @@
+# Validation
+
+This document can be used to validate the state overrides and diffs resulting from the execution of the FP upgrade transaction.
+
+For each contract listed in the state diff, please verify that no contracts or state changes shown in the Tenderly diff are missing from this document. Additionally, please verify that for each contract:
+
+- The following state changes (and none others) are made to that contract. This validates that no unexpected state changes occur.
+- All addresses (in section headers and storage values) match the provided name, using the Etherscan and Superchain Registry links provided. This validates the bytecode deployed at the addresses contains the correct logic.
+- All key values match the semantic meaning provided, which can be validated using the storage layout links provided.
+
+## Nested Safe State Overrides and Changes
+
+This task is executed by the nested 2/2 `ProxyAdminOwner` Safe. Refer to the
+[generic nested Safe execution validation document](../../../NESTED-VALIDATION.md)
+for the expected state overrides and changes.
+
+The `approvedHashes` mapping **key** of the `ProxyAdminOwner` that should change during the simulation is
+- Council simulation: `0x1188ef4fb2355c49d442a7f950dad4492a2961a34786ce407c3db8c8d331250d`
+- Foundation simulation: `0x030c89f6f0b759d892697ad05de88183a3f6f579e0abe30c81427eaefb20a459`
+
+calculated as explained in the nested validation doc:
+```sh
+SAFE_HASH=0x572c29de03270811431295a31c25273c7dbf3bc0f17f29a8fe3131f86b7a9cb6 # "Nested hash:"
+SAFE_ROLE=0xf64bc17485f0B4Ea5F06A96514182FC4cB561977 # Council
+cast index bytes32 $SAFE_HASH $(cast index address $SAFE_ROLE 8)
+# 0x1188ef4fb2355c49d442a7f950dad4492a2961a34786ce407c3db8c8d331250d
+
+SAFE_ROLE=0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B # Foundation
+cast index bytes32 $SAFE_HASH $(cast index address $SAFE_ROLE 8)
+# 0x030c89f6f0b759d892697ad05de88183a3f6f579e0abe30c81427eaefb20a459
+```
```

Comment:
same as above
```suggestion
calculated as explained in the nested validation doc:
```sh
SAFE_HASH=0x572c29de03270811431295a31c25273c7dbf3bc0f17f29a8fe3131f86b7a9cb6 # "Nested hash:"
SAFE_SIGNER=0xf64bc17485f0B4Ea5F06A96514182FC4cB561977 # Council
cast index bytes32 $SAFE_HASH $(cast index address $SAFE_SIGNER 8)
# 0x1188ef4fb2355c49d442a7f950dad4492a2961a34786ce407c3db8c8d331250d

SAFE_SIGNER=0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B # Foundation
cast index bytes32 $SAFE_HASH $(cast index address $SAFE_SIGNER 8)
# 0x030c89f6f0b759d892697ad05de88183a3f6f579e0abe30c81427eaefb20a459
```
```

Reply:
renamed to Safe Signer